### DISCUSSION TITLE
useForm returnValue cause infinity loop

### DISCUSSION THREAD

**Discussion by vaynevayne** [mantinedev/mantine#4277] [Q&A]
<img width="720" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/7213994e-99f5-47b5-adfc-3ebe3dd0aade">
<img width="658" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/81285e46-f80e-40a7-bca3-b123320feb66">

when is add form to dependencies, cause infinity loop
<img width="591" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/a5046b8f-295a-4862-a15b-949cb5e80216">

**Comment by vaynevayne** [comment]
Can it be changed like this?

<img width="639" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/99c4eae2-b405-4274-93db-bc95f352f8b0">

// or 
<img width="506" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/a17fe4d8-cb70-40e6-87a6-97823be86de0">

**Comment by rtivital** [comment]
Do not add form to useEffect dependencies
